app/api/subscribe/route.ts

```diff
@@ -1,6 +1,7 @@
 import { main } from "@/lib/Server";
+import { Request } from "@/lib/Services/Request";
 import { Effect } from "effect";
 
 export async function POST(request: Request): Promise<Response> {
-  return main(request).pipe(Effect.runPromise);
+  return main.pipe(Effect.provideService(Request, request), Effect.runPromise);
```

Comment:
Moving Request to context provides a bit more flexibility.

Reply:
Would it make sense to implement a custom `Runtime`? How would it work?

Reply:
I'm not sure if next.js let's you reuse server side variables. If they are using lambdas then probably not.

But if you can share some context between http requests, then yes you could use a custom runtime.